File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
float l = 0;
float w = 0;
float s = 0;
float index = 0;

int main(void)
{
    string text = get_string("Text: \n");
    l = count_letters(text);
    w = count_words(text);
    s = count_sentences(text);
    //printf("%f letters , %f words , %f sentences \n",l,w,s);
    float li = l / w * 100;
    float se = s / w * 100;
    index = 0.0588 * li - 0.296 * se - 15.8;
    index = round(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)index);
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            letters++;
        }
    }
    return letters;
}
int count_words(string text)
{
    int words = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 32)
        {
            words++;
        }
    }
    return words;
}
int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if ((text[i] == 46) || (text[i] == 63) || (text[i] == 33))
        {
            sentences++;
        }
    }
    return sentences;

}

File: sentimental-readability/readability.py
from cs50 import get_string

input = get_string("Text : ")

l = 0
w = 1
s = 0
for i in range(len(input)):
    if input[i] == " ":
        w += 1
    elif input[i] == "." or input[i] == "!" or input[i] == "?":
        s += 1
    elif input[i].isalpha():
        l += 1
lw = l / w * 100
se = s / w * 100
cli = 0.0588 * lw - 0.296 * se - 15.8
# print(cli)
rcli = round(cli)
# print(rcli)
if rcli < 1:
    print("Before Grade 1")
elif rcli > 16:
    print("Grade 16+")
else:
    print("Grade ", rcli)